File: truth_set/javascript/logic/performance/processData.js
function processData(data) {
    const newData = [...data];
    for (let i = 0; i < newData.length; i++) {
        newData[i] *= 2;
    }
    return newData;
}

const numbers = Array.from({ length: 1000000 }, (_, i) => i);
const processedArray = processData(numbers);
\ No newline at end of file

File: truth_set/go/logic/performance/processData.go
package performance

func processData(data []int) []int {
    newData := make([]int, len(data))
    copy(newData, data)
    
    for i := range newData {
        newData[i] *= 2
    }
    return newData
}

func processDataMain() {
    numbers := make([]int, 1000000)
    for i := range numbers {
        numbers[i] = i
    }
    
    processedArray := processData(numbers)
    _ = processedArray
}
\ No newline at end of file